src/msw/treectrl.cpp

```diff
@@ -3925,8 +3928,9 @@ void wxTreeCtrl::DoFreeze()
 
 void wxTreeCtrl::DoThaw()
 {
-    if (!m_pendingFreeze)
+    if ( !m_pendingFreeze )
         wxTreeCtrlBase::DoThaw();
+    //else: we never froze the control in the first place
```

Comment:
Need to reset m_pendingFreeze to false in here. This was not necessary before, but now the code is relying solely on this variable. Consider of following (unlikely) scenario:
```c++
DeleteAllItems();
Freeze(); // Pending freeze true (no items)
Thaw(); 
AddRoot(); // Pending freeze still true because it never actually got frozen
```